src/wireviz/bom_helper.py

```diff
@@ -0,0 +1,42 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+
+import csv
+from wireviz import wv_helper
+from wireviz.wv_helper import open_file_write
+
+EXCEL_CSV = csv.excel
+EXCEL_TSV = csv.excel_tab
+UNIX_CSV = csv.unix_dialect
+WIREVIZ_TSV = type('Wireviz BOM', (csv.Dialect, object), dict(
+    delimiter='\t',
+    doublequote=True,
+    escapechar=None,
+    lineterminator='\n',
+    quoting=0,
+    skipinitialspace=False,
+    strict=False,
+    quotechar='"'
+))
+csv.register_dialect('Wireviz BOM', WIREVIZ_TSV)
+
+_csv_formats = { EXCEL_CSV, UNIX_CSV }
+_tsv_formats = { EXCEL_TSV, WIREVIZ_TSV }
+
+_csv_ext = '.bom.csv'
+_tsv_ext = '.bom.tsv'
+
+def generate_bom_outputs(base_filename, bomdata, *argv):
+    expanded_csv_names =  len(_csv_formats.intersection(set(argv))) > 1
+    expanded_tsv_names = len(_tsv_formats.intersection(set(argv))) > 1
+    for fmt in argv:
+        if fmt in _csv_formats:
+            file = csv.writer(open_file_write(base_filename + ("_" + fmt.__name__ if expanded_csv_names else "") + _csv_ext, fmt.lineterminator), fmt)
+
+        elif fmt in _tsv_formats:
+            file = csv.writer(open_file_write(base_filename + ("_"+fmt.__name__ if expanded_tsv_names else "") + _tsv_ext, fmt.lineterminator), fmt)
+        else:
+            raise KeyError("Unknown BOM Format Specified")
+        file.writerows(wv_helper.flatten2d(bomdata))
```

Comment:
The `*argv` parsing (lines 29 to 31) are not very pythonic, I'd request you change it.
The `len(intersection(set())) > 1` is rather cryptic... and the use case of calling this function requesting multiple CSV and TSV formats at once is pretty much an edge case IMHO... especially since currently there are set defaults, inaccessible to the user.

My proposal: Call the function with a list as the third argument:
```
def generate_bom_outputs(base_filename, bomdata, formats):
    if not isinstance(formats, List):
        formats = [formats]
```
and then iterate over the list.
It could then be called using
```
bom_helper.generate_bom_outputs(filename,bom_list,[bom_helper.WIREVIZ_TSV, bom_helper.EXCEL_CSV])
```

Reply:
That could work well! I will also check briefly to see if the Excel TSV format is essentially compatible with the one wireviz natively generates (and probably have the OS choose if new lines are unix or windows style), and then the csv formats can be directly passed through.

Reply:
Also, excuse any python style issues, I will try to fix them whenever pointed out. I'm a Java / C / MATLAB developer, primarily, who has wanted to learn python for far too long!

Reply:
> I'm a Java / C / MATLAB developer, primarily, who has wanted to learn python for far too long!

You're not alone, I've done mostly embedded programming in C++ / Arduino before.
I'm loving Python's readability and trying to keep WireViz as pythonic and understandable as possible :)